packages/features/portfolio/src/hooks/use-portfolio-snote-table.tsx

```diff
@@ -0,0 +1,187 @@
+import { useState, useMemo } from 'react';
+import { FormattedMessage } from 'react-intl';
+import {
+  MultiValueIconCell,
+  DataTableColumn,
+  MultiValueCell,
+  DisplayCell,
+} from '@notional-finance/mui';
+import { TotalEarningsTooltip } from '../components';
+import {
+  useAccountDefinition,
+  useFiat,
+  useStakedNOTEPoolReady,
+  useStakedNoteData,
+} from '@notional-finance/notionable-hooks';
+import { ExpandedState } from '@tanstack/react-table';
+import { useTheme } from '@mui/material';
+import { Network, lastValue, PORTFOLIO_ACTIONS } from '@notional-finance/util';
+import { Registry, TokenBalance } from '@notional-finance/core-entities';
+import { formatNumberAsPercentWithUndefined } from '@notional-finance/helpers';
+import { useHistory, useLocation } from 'react-router';
+
+export function usePortfolioSNOTETable() {
+  const theme = useTheme();
+  const history = useHistory();
+  const { pathname: currentPath } = useLocation();
+  const [expandedRows, setExpandedRows] = useState<ExpandedState>({});
+  const account = useAccountDefinition(Network.mainnet);
+  const snoteBalance = account?.balances.find((t) => t.symbol == 'sNOTE');
+  const isPoolReady = useStakedNOTEPoolReady();
+  const sNOTEPool = Registry.getExchangeRegistry().getSNOTEPool();
```

Comment:
Only run this when `isPoolReady` is true:
`const sNOTEPool = isPoolReady ? Registry.getExchangeRegistry().getSNOTEPool() : undefined`